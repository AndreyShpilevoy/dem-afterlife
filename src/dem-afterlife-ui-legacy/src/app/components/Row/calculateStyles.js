const calculateStyles = () => ({
    row: {
        width: '100%',
        boxSizing: 'border-box',
        display: 'flex',
        flex: '0 1 auto',
        flexDirection: 'row',
        flexWrap: 'wrap',
        '&.reverse': {
            flexDirection: 'row-reverse'
        }
    }
});

export default calculateStyles;