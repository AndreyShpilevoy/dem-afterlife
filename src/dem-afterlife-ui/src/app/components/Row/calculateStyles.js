const calculateStyles = ({themeName}) => ({
    row: {
        width: '100%',
        boxSizing: 'border-box',
        display: 'flex',
        flex: '0 1 auto',
        flexDirection: 'row',
        flexWrap: 'wrap',
        '&.reverse': {
            flexDirection: 'row-reverse'
        }
    },
    options: {meta: 'Row', themeName}
});

export default calculateStyles;