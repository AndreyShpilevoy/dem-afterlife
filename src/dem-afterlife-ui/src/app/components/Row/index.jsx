import React from 'react';
import {bool, node, string, func, shape} from 'prop-types';
import {css, withStyles} from 'styles';
import {defaults} from 'utils';
import calculateStyles from './calculateStyles';

const constructClassNameString = (reverse, styles) =>
    css([styles.row, reverse ? 'reverse' : defaults.emptyString]);

export const RowPure = ({reverse, children, styles, className, onClick}) => {
    const localeClassName = `${constructClassNameString(reverse, styles)} ${className || defaults.emptyString}`;
    return (
        <div className={localeClassName} onClick={onClick}>
            {children}
        </div>
    );
};

RowPure.propTypes = {
    className: string,
    styles: shape().isRequired,
    children: node.isRequired,
    reverse: bool,
    onClick: func
};

RowPure.defaultProps = {
    className: defaults.emptyString,
    onClick: defaults.emptyFunction,
    reverse: false
};

export default withStyles(theme => calculateStyles(theme))(RowPure);