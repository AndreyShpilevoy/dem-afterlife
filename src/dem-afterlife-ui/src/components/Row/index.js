import React from "react";
import { bool, node, string, func, shape } from "prop-types";
import { concatStyleNames, injectSheet } from "styles";
import { handleEnterKeyDown } from "utils";
import calculateStyles from "./calculateStyles";

const constructClassNameString = (reverse, classes) =>
  concatStyleNames([classes.row, reverse ? "reverse" : ""]);

export const RowPure = ({ reverse, children, classes, className, onClick }) => {
  const localClassName = `${constructClassNameString(
    reverse,
    classes
  )} ${className}`;
  const handleKeyDown = event =>
    handleEnterKeyDown(event) ? onClick() : () => ({});
  return onClick ? (
    <div
      className={localClassName}
      onClick={onClick}
      onKeyDown={handleKeyDown}
      role="button"
      tabIndex={0}
    >
      {children}
    </div>
  ) : (
    <div className={localClassName}>{children}</div>
  );
};

RowPure.propTypes = {
  className: string,
  classes: shape().isRequired,
  children: node.isRequired,
  reverse: bool,
  onClick: func // eslint-disable-line react/require-default-props
};

RowPure.defaultProps = {
  className: "",
  reverse: false
};

export default injectSheet(calculateStyles, { componentName: "Row" })(RowPure);
