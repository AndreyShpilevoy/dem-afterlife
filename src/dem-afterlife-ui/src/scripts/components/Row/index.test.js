/* eslint no-undef: 'off'*/
/* eslint import/no-extraneous-dependencies: 'off' */

import React from 'react';
import {shallow} from 'enzyme';
import Row, {RowPure} from './index';

describe('Row', () => {
    const hocProps = {
        theme: 'default',
        classNames:
        {
            '.row-0-0.reverse': '.row-0-0.reverse-0-1',
            row: 'row-0-0'
        }
    };

    it('HOC match expected snapshot', () => {
        expect(shallow(<Row theme={hocProps.theme} />)).toMatchSnapshot();
    });

    it('Pure component match expected snapshot', () => {
        expect(shallow(<RowPure classNames={hocProps.classNames}/>)).toMatchSnapshot();
    });

    it('Pure component match expected snapshot with reverse prop', () => {
        expect(shallow(<RowPure classNames={hocProps.classNames} reverse/>)).toMatchSnapshot();
    });
});
