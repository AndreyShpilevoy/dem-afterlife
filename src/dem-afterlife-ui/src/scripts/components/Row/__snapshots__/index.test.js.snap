// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row HOC match expected snapshot 1`] = `
<RowPure
  classNames={
    Object {
      ".row-0-0.reverse": ".row-0-0.reverse-0-1",
      "row": "row-0-0",
    }
  }
  theme="default"
/>
`;

exports[`Row Pure component match expected snapshot 1`] = `
<div
  className="row-0-0"
/>
`;

exports[`Row Pure component match expected snapshot with reverse prop 1`] = `
<div
  className="row-0-0 reverse"
/>
`;
